pallets/msa/src/lib.rs

```diff
@@ -218,13 +226,25 @@ pub mod pallet {
 	#[pallet::storage]
 	#[pallet::getter(fn get_payload_signature_registry)]
 	pub(super) type PayloadSignatureRegistry<T: Config> = StorageDoubleMap<
-		_,
-		Twox64Concat,
+		_,                      // prefix
+		Twox64Concat,           // hasher for key1
 		T::BlockNumber, // Bucket number. Stored as BlockNumber because I'm done arguing with rust about it.
-		Twox64Concat,
+		Twox64Concat,   // hasher for key2
 		MultiSignature, // An externally-created Signature for an external payload, provided by an extrinsic
 		T::BlockNumber, // An actual flipping block number.
-		OptionQuery,
+		OptionQuery,    // The type for the query
+		GetDefault,     // OnEmpty return type, defaults to None
+		T::MaxSignaturesStored, // Maximum total signatures to store
+	>;
+
+	/// Records how many signatures are currently stored in each virtual signature registration bucket
+	#[pallet::storage]
+	pub(super) type PayloadSignatureBucketStorageCount<T: Config> = StorageMap<
+		_,
```

Comment:
PayloadSignatureBucketStorageCount > I think we can drop storage from the name since the macro above implies its storage.